test/auction/AuctionOpeningTest.t.sol

```diff
@@ -0,0 +1,432 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import "../utils/BaseTest.t.sol";
+
+contract AuctionsOpeningTest is BaseTest {
+    uint256 constant STEPS = 10;
+
+    function initBalances(address firstUser, address secondUser) public {
+        deal(address(collateralToken), address(this), 100 ether);
+        deal(address(borrowToken), address(this), 100 ether);
+
+        deal(address(collateralToken), firstUser, 100 ether);
+        deal(address(borrowToken), firstUser, 100 ether);
+
+        deal(address(collateralToken), secondUser, 100 ether);
+        deal(address(borrowToken), secondUser, 100 ether);
+    }
+
+    function depositAndExecuteAuction(address user) public {
+        vm.startPrank(user);
+        borrowToken.approve(address(ltv), 1 ether);
+        ltv.deposit(1 ether, user);
+        vm.stopPrank();
+
+        int256 futureBorrow = ltv.futureBorrowAssets();
+        uint256 amount = futureBorrow > 0 ? uint256(futureBorrow) : uint256(-futureBorrow);
+        collateralToken.approve(address(ltv), amount);
+        ltv.executeAuctionBorrow(-futureBorrow);
+    }
+
+    function initTest() public returns (address, address) {
+        address firstUser = address(0x123);
+        address secondUser = address(0x321);
+        initBalances(firstUser, secondUser);
+        depositAndExecuteAuction(firstUser);
+
+        return (firstUser, secondUser);
+    }
+
+    // grows for borrow:
+
+    function checkMaxWithdrawGrowsDuringAuctionOpening(address user) public {
+        uint256 prevMaxWithdraw = ltv.maxWithdraw(user);
+        uint256 currentMaxWithdraw;
+        uint256 prevPreviewWithdraw = ltv.previewWithdraw(prevMaxWithdraw);
+        uint256 currentPreviewWithdraw;
+
+        for (uint256 i = 1; i <= STEPS; i++) {
+            vm.roll(block.number + 1);
```

Comment:
here in every test we need to test token price growth also - in some specific percentage every iteration (with some delta)